src/Analyser/Analyser.php

```diff
@@ -153,33 +153,41 @@ public function analyse(
 
 					$ignoreComments = new IgnoreCommentsCollection();
 
-					$scopeBenchmarkTime = $this->benchmarkStart();
 					$this->nodeScopeResolver->processNodes(
 						$parserNodes,
 						$this->scopeFactory->create(ScopeContext::create($file)),
-						function (\PhpParser\Node $node, Scope $scope) use (&$fileErrors, $file, &$scopeBenchmarkTime, $ignoreComments): void {
-							$this->benchmarkEnd($scopeBenchmarkTime, 'scope');
-							$uniquedAnalysedCodeExceptionMessages = [];
-
+						function (\PhpParser\Node $node, Scope $scope) use (&$fileErrors, $file, $ignoreComments): void {
 							if (
-								!$node instanceof VirtualNode
-								&& !$node instanceof Expression
-								&& $scope->isInFirstLevelStatement()
+								$node instanceof VirtualNode
+								|| $node instanceof Expression
+								|| !$scope->isInFirstLevelStatement()
 							) {
-								foreach ($node->getComments() as $comment) {
-									try {
-										$ignoreComment = $this->commentParser->parseIgnoreComment($comment, $node);
-									} catch (\Nette\Utils\RegexpException $e) {
-										$fileErrors[] = new Error($e->getMessage(), $file, $comment->getLine(), false);
-										continue;
-									}
-									if ($ignoreComment === null) {
-										continue;
-									}
+								return;
+							}
 
-									$ignoreComments->add($ignoreComment);
+							foreach ($node->getComments() as $comment) {
+								try {
+									$ignoreComment = $this->commentParser->parseIgnoreComment($comment, $node);
+								} catch (\Nette\Utils\RegexpException $e) {
+									$fileErrors[] = new Error($e->getMessage(), $file, $comment->getLine(), false);
+									continue;
 								}
+								if ($ignoreComment === null) {
+									continue;
+								}
+
+								$ignoreComments->add($ignoreComment);
 							}
+						}
+					);
+
+					$scopeBenchmarkTime = $this->benchmarkStart();
+					$this->nodeScopeResolver->processNodes(
```

Comment:
I don't understand this commit. Why are we calling `$this->nodeScopeResolver->processNodes()` twice? This will lead to a huge performance impact.